service/src/main/java/uk/gov/hmcts/reform/fpl/service/CaseService.java

```diff
@@ -0,0 +1,137 @@
+import com.google.common.base.Strings;
+import com.google.common.collect.Maps;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.MediaType;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+import org.springframework.web.multipart.MultipartFile;
+import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.reform.ccd.client.CoreCaseDataApi;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDataContent;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.ccd.client.model.Event;
+import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
+import uk.gov.hmcts.reform.document.DocumentUploadClientApi;
+import uk.gov.hmcts.reform.document.domain.Document;
+import uk.gov.hmcts.reform.document.domain.UploadResponse;
+import uk.gov.hmcts.reform.document.utils.InMemoryMultipartFile;
+import uk.gov.hmcts.reform.fpl.templates.DocumentTemplates;
+import uk.gov.hmcts.reform.pdf.generator.HTMLTemplateProcessor;
+import uk.gov.hmcts.reform.pdf.generator.HTMLToPDFConverter;
+import uk.gov.hmcts.reform.pdf.generator.PDFGenerator;
+import uk.gov.hmcts.reform.pdf.generator.XMLContentSanitizer;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.google.common.collect.Lists.newArrayList;
+
+
+@Component
+public class CaseService {
+
+    public static final String JURISDICTION_ID = "PUBLICLAW";
+    public static final String CASE_TYPE = "Shared_Storage_DRAFTType";
+    private final HTMLToPDFConverter converter = new HTMLToPDFConverter();
+
+    @Autowired
+    private DocumentTemplates documentTemplates;
+    @Autowired
+    private DocumentUploadClientApi documentUploadClient;
+    @Autowired
+    private CoreCaseDataApi coreCaseDataApi;
+    @Autowired
+    private AuthTokenGenerator authTokenGenerator;
+
+    /**
+     *
+     * @param authorization
+     * @param userId
+     * @param request
+     * @throws JSONException
+     * @throws IOException
+     *
+     * Converts request @param to a map that contains the case data and the headers that can be used to create the pdf and view contents of the
+     * callback
+     */
+    @Async
+    @SuppressWarnings("unchecked")
+    public void handleCaseSubmission(
+        String authorization,
+        String userId,
+        CallbackRequest request) throws JSONException, IOException {
+        byte[] template = documentTemplates.getHtmlTemplate();
+
+        ObjectMapper mapper = new ObjectMapper();
+        Map<String, Object> map = mapper.convertValue(request.getCaseDetails(), Map.class);
+
+        System.out.println("map = " + map);
+        byte[] pdfDocument = converter.convert(template, map);
```

Comment:
Please extract operate class that is responsible for PDF generation, something we can unit test